odl/solvers/advanced/chambolle_pock.py

```diff
@@ -0,0 +1,284 @@
+        Controls the extent to which ``proximal_primal`` maps points
+        towards the minimum of G.
+    sigma : positive `float`
+        Step size parameter for the update of the dual variable y. Controls
+        the extent to which ``proximal_dual`` maps points towards the
+        minimum of F_cc.
+    proximal_primal : `callable`
+        Evaluated at ``tau``, the function returns the proximal operator,
+        prox_tau[G](x), of the functional G. The domain of G and its
+        proximal operator instance are the space, X, of the primal variable
+        x  i.e. the domain of ``op``.
+    proximal_dual : `callable`
+        Evaluated at ``sigma``, the function returns the proximal operator,
+        prox_sigma[F_cc](x), of the convex conjugate, F_cc, of the function
+        F. The domain of F_cc and its proximal operator instance are the
+        space, Y, of the dual variable y i.e. the range of ``op``.
+    theta : `float` in [0, 1], optional
+        Relaxation parameter
+    gamma : positive `float`
+        Acceleration parameter. If not `None` overwrites ``theta`` and uses
+        variable relaxation parameter and step sizes with ``tau`` and
+        ``sigma`` as initial values. Requires G or F_cc to be uniformly convex.
+    niter : non-negative `int`, optional
+        Number of iterations
+    partial : `Partial`, optional
+        If not `None` the `Partial` instance(s) are executed in each
+        iteration, e.g. plotting each iterate
+    x_relax : element in the domain of ``op``, optional
+        Required to resume iteration. If `None` it is a copy of the primal
+        variable x.
+    y : element in the range of ``op``, optional
+        Required to resume iteration. If `None` it is set to a zero element
+        in Y which is the range of ``op``.
+
+    References
+    ----------
+    .. [CP2011a] `Chambolle, Antonin and Pock, Thomas. A First-Order
+     Primal-Dual Algorithm for Convex Problems with Applications to Imaging.
+     J. Math. Imaging Vis. 40 120, 2011.
+     <http://dx.doi.org/10.1007/s10851-010-0251-1>`_
+
+    .. [CP2011b] `Chambolle, Antonin and Pock, Thomas. Diagonal
+     preconditioning for first order primal-dual algorithms in convex
+     optimization. 2011 IEEE International Conference on Computer Vision
+     (ICCV), 1762, 2011. <http://dx.doi.org/10.1109/ICCV.2011.6126441>`_
+
+    .. [SJP2012] `Sidky, Emil Y, Jorgensen, Jakob H, and Pan, Xiaochuan. Convex
+     optimization problem prototyping for image reconstruction in computed
+     tomography with the Chambolle-Pock algorithm, Phys Med Biol, 57 3065,
+     2012. <http://stacks.iop.org/0031-9155/57/i=10/a=3065>`_
+
+    .. [PB2014] Parikh, Neal and Boyd, Stephen. *Proximal Algorithms*.
+     Foundations and Trends in Optimization 1 127, 2014.
+     http://stacks.iop.org/0031-9155/57/i=10/a=3065
+
+    .. [R1970] Rockafellar, R. Tyrrell. *Convex analysis*. Princeton University
```

Comment:
Good citation style in general. I made an issue #279 on this so we have uniform style across. I wouldn't include links.